### DISCUSSION TITLE
High power consumption of the green vs. the purple Nano style LQFP32 MiniEVB

### DISCUSSION THREAD

**Discussion by wollewald** [dbuezas/lgt8fx#281] [General]
I am currently testing different LGT8F328P boards and found a strange behavior of the green Nano style LQFP32 MiniEVB. I uploaded the bare minimum sketch and chose 32 MHz clock speed without clock divider. Then I disconnected the board from USB and connected it board via GND / VCC with 5V using a stable and potent power supply. 

- With this setup, I measured a current of 32 mA. 
- If I do the same with the purple Nano style LQFP32 MiniEVB I measure a current of 16.3 mA! 
- At 3 V the green board consumes 16.7 mA  vs. 13.4 mA (purple board), which is much more similar

Since I found this quite strange, I conducted a further test with the green board. Again, I supplied it with 5 V via GND / VCC. Then I reduced the voltage stepwise and measured the current. I found:

5.0V - 32.0 mA
4.9 V - 31.4 mA 
4.8 V - 30.9 mA
4.7 V - 30.4 mA
4.6 V - 30.1 mA
4.5 V - 29.9 mA
4.4. V - 29.5 mA
4.3 V - 20.2 mA (!)
4.2 V - 19.9 mA
4.1 V - 19.6 mA
.....

Any idea what's happening between 4.4 V and 4.3 V? And why does the purple board behave so much different. Could the reason be related to different USB to TTL chips? The green one uses a HT42B534-2 and the purple one a CH9340C. Does the HT42B534-2 consume much more power than the CH9340C and switches off a 4.3 V? Very much speculation.

**Comment by LaZsolt** [comment]
| Operating supply current: | - |
| - | - |
| CH9340C | 7-15 mA |
| HT42B534-2 | 11-16 mA |

No big difference in active mode.
In my opinion around 5V HT42B534-2 remains active even when it is not powered by USB. Maybe it is a board design flaw.

**Comment by dbuezas** [comment]
If the serial chip is power from the regulator instead of the usb-5v (before the usb diode), then that 4.3v threshold may be when the LDO runs out of headroom.
Have you measured what voltage you get after the LDO at that input-voltage? 
Hypothesis: It drops a lot, the usb chip shuts down, and the lgt still runs (because it is happy at 1.8v)

**Comment by wollewald** [comment]
Maybe my description was a bit misleading. I have connected the power supply to the 5V pin which **_is_** VCC otf the LGT8F328P. So, I am not going through VIN and the LDO. This is not recommended, but only for safety reason and you need a stable power supply.

So why am I doing this all? It's a nice feature of the LGT8F328P that you can run it at 32 MHz down to 1.8 V. Therefore you can directly supply it e.g. with a Li-Ion battery. 

I have sacrifed one of the green Nano LQFP32 MiniEVBs by cutting the GND of the HT42B534-2. With this setup the power consumption goes down 16-17 mA at 5 V power supply through the 5V pin. Overall this is not very surprising since it is effectively the same configuration as you have on a Pro Mini style MiniEVB. And according to the table:
https://github.com/dbuezas/lgt8fx#power-consumption--5v
this is a similar power consumption. 

OK, so far this all makes sense but still there is the question why the purple Nano LQFP32 MiniEVB behaves differently. But I think I have found the answer myself. According to the data sheet of the CH9340 the power consumption is very low when disconnected from USB:

![Screenshot 2023-05-28 110151](https://github.com/dbuezas/lgt8fx/assets/41305162/fba326c9-2ff2-4833-b510-38fb22f77214)

On the other hand, according to the HT42B534-2 datasheet it should also use low power when disconnected from USB:

![Screenshot 2023-05-28 113950](https://github.com/dbuezas/lgt8fx/assets/41305162/e654eabb-cc1f-4e0b-8fed-e9ef5dd4a202)

So, somehow that not working. However, to complete the confusion, there is a way to significantly reduce the power consumption of the green LQFP32 MiniEVB. If you power it up with a low voltage supply (e.g. 5V to VIN or 3V to the 5V pin/VCC) and then increase the voltage to let's say 8V at VIN or 5V at the 5V pin the power consumption (still at 32 MHz) is roughly 22 mA. This is reproducible. 

For the LQFP48 MiniEVB I found the same behavior like for the purple LQFP32 MiniEVB, which means relatively low power consumption at 5V power supply directly to 5V / VCC. This makes sense again because it uses a CH340 (without the "9"). In its data sheet I found:

![Screenshot 2023-05-28 112856](https://github.com/dbuezas/lgt8fx/assets/41305162/877cc85f-5927-4edf-a46e-193cae52d091)

So, in summary my conclusion is: there is an issue with the HT42B534-2 (at least with the boards I tested) when disconnected from USB. It's still active although it shouldn't. People who want to use the Nano LQFP32 MiniEVB with battery power supply should either use the purple version or use a voltage below ~4.4 V. 

I hope you can follow my trains of thought!

**Reply by dbuezas** [reply]
Makes sense. For battery powered, I usually completely remove the regulator too. I think they still draw current even when theres voltage only in the output, and if deep sleep is used, this (i think) makes a big difference.

Btw, now that there are sleep modes, we could extend that table with them :). I don't have access to my electronics at the moment

**Comment by wollewald** [comment]
I will try to remove the voltage regulator. Good point. 

I am current looking into the sleep modes and will make some measurements. Moreover, the results of the purple Nano could be added. Will need a bit to complete.

**Comment by trimarco232** [comment]
Hi ,
imho , powering the usb chip otherwise than direct by usb is just an old stupidity

**Reply by dbuezas** [reply]
Yeah, I don't get that. There's even a diode on the usb positive rail already. Maybe the chip misbehaves it gets voltage on its RX pin only

**Reply by wollewald** [reply]
@trimarco232 I agree that powering the USB chip not by USB makes no sense - in case you want to use it all. But if you use the lgt8f328p module in a battery powered project you usually power it via VIN or VCC. In this case it's a side effect that you also power the USB chip. It shouldn't be a problem, since both USB chips in question should swith off. The ch9340 does this correctly, the HT42B534-2 doesn't. The effect is that the green boards consume much more current than necessary.

**Reply by trimarco232** [reply]
"Maybe the chip misbehaves it gets voltage on its RX pin only"
or it gets its power trough its RX pin (protection diodes) , and this is precisely not what we don't want
but one should be able to let the Tx pin of the mcu high Z , in order to avoid this ...

"the HT42B534-2 doesn't"
there is interesting stuff in its datasheet : 
`UART Power Down and Wake-up
If the USB host sends a suspend signal to the 
HT42B534-x USB device, it will enter the suspend 
mode. It is recommended to ensure that the UART 
data transmission or reception has been finished 
before the device enters the suspend mode.
The UART function contains a receiver RX pin wakeup function. A falling edge on the RX pin will wake 
up the device from the suspend mode.`

**Reply by wollewald** [reply]
Not sure how to send a suspend signal to the HT42B534 from the host, when there is no host. On the other hand the data sheet says:

_If there is no signal on the USB bus for over 3ms, the 
USB device will enter the suspend mode. The device 
enters the suspend state to meet the requirements of 
the USB suspend current specification._

That means for me that the HT42B534 should switch off (suspend mode) as long as there is no regular falling edge on RX (and I did not find that). 

But what I found is the following: If you pull pown D- (I took a 10kOhm resistor) while you power on via VIN, the HT42B534 goes into suspend mode. The current goes down to ~16.9 mA. Solves the issue in theory but is not a convenient solution since you need to solder a connection to the D- Pin on the board or use something like this:

![Screenshot 2023-06-11 121051](https://github.com/dbuezas/lgt8fx/assets/41305162/e7fe36f6-980b-4ea4-97ab-c02b51931c42)

Would be great if someone finds a better solution.

If you pull down D- after the module has been powered on the current drops from 32 mA to 22 mA.  Perhaps this information will help those who want to delve a little deeper. 

I will stop my investigations here. My conclusion is that people should take the purple module if they want to use an LQFP32 LGT8F328P in battery powered projects. Or, even better, use the "Pro Mini style" board or the bare LGT8F328P IC.

**Comment by wollewald** [comment]
I close this discussion. The issue and how solve it is described here:
https://wolles-elektronikkiste.de/en/minievb-boards-an-overview#reduce_pwr_cons_green_nano_32_board
